2199/3/controllers/application_controller.rb

```diff
@@ -0,0 +1,38 @@
+require 'sinatra'
+require 'pry'
+# class ApplicationController
+class ApplicationController < Sinatra::Base
+  set views: 'views/'
+
+  get '/' do
+    redirect '/pages'
+  end
+
+  get '/pages' do
+    @pages = Page.all
+    erb :index
+  end
+
+  get '/pages/new' do
+    erb :new
+  end
+
+  get '/pages/:id' do
+    @page = Page[params[:id]]
+    erb :show
+  end
+
+  post '/pages' do
+    link = params[:link]
+    @page = Page.find(link: link).first
+    unless @page
+      comment_texts = OnlinerPageParser.new(link).top_comment_texts
+      comments_with_score = CommentAnalyzer.new(comment_texts).analyze
```

Comment:
Eventually, it's a `Comment` or `Comments` analyzer? From your code, I see that it analyzes all comments, not the particular one.